examples/pagerank_example.py

```diff
@@ -0,0 +1,79 @@
+import time
+
+import sparse
+
+import networkx as nx
+from networkx.algorithms.link_analysis.pagerank_alg import _pagerank_scipy
+
+import numpy as np
+import scipy.sparse as sp
+
+
+def pagerank(G, alpha=0.85, max_iter=100, tol=1e-6) -> dict:
+    N = len(G)
+    if N == 0:
+        return {}
+
+    alpha = sparse.asarray(alpha)
+    nodelist = list(G)
+    A = nx.to_scipy_sparse_array(G, dtype=float, format="csc")
+    A = sparse.asarray(A)
+    S = sparse.sum(A, axis=1)
+    S = sparse.where(sparse.asarray(0.0) != S, sparse.asarray(1.0) / S, S)
+
+    # TODO: spdiags https://github.com/willow-ahrens/Finch.jl/issues/499
+    Q = sparse.asarray(sp.csc_array(sp.spdiags(S.todense(), 0, *A.shape)))
+    A = Q @ A
```

Comment:
IIUC This is equivalent to https://data-apis.org/array-api/latest/extensions/generated/array_api.linalg.diagonal.html#diagonal, which is [supported by the Numba backend](https://github.com/pydata/sparse/blob/400d7f0b52042ae21c49a7e30aea19b3d1198e47/sparse/numba_backend/_coo/common.py#L804). Let's use that instead.

```suggestion
    Q = sparse.diagonal(S)
    A *= Q
```

Reply:
For Finch, we can add support for it later, as the example only needs to work on Numba for now.

Reply:
You might need `Q[None, :]` to match the results, but otherwise this is equivalent.

Reply:
But the `pagerank` example is run either with Finch or Numba backend. when we run with Finch backend (and I think it should be runnable with Finch) then we can't execute one line with Numba.

Reply:
I would prefer to keep these examples Finch and Numba ready, and update implementation once both backends support something new.

Reply:
> I would prefer to keep these examples Finch and Numba ready, and update implementation once both backends support something new.

According to the deliverable, we just need a script that will demonstrate a speedup against the "old" code. If you would prefer, we can always keep this PR open until `xp.diagonal` is supported by Finch.